### PR TITLE
 Adds and modifies abductor glands 

### PR DESCRIPTION
**Pull request by XDTM**
## About The Pull Request

Adds three new abductor glands and makes minor changes to existing ones.
Full change list:
- New gland: [Blood] Periodically randomizes the owner's bloodtype into a random reagent.
- New gland: [Quantum] Periodically links with a random person in view, then later swaps the positions of the owner and the other person.
- New gland: [Access] After it activates, makes the owner have intrinsic all access.
- Fixed a bug where abductor scientists could not see the "real" gland names.
- Mind control on the Mindshock gland now broadcasts the mind control to everyone in view _without_ affecting the original target.
- Mind control on the Quantum gland will also affect the linked target if it currently has one.
- Changed gland icons so most icons used are shared by two gland types, making it harder to tell which type it is to an untrained eye.
- Reagent gland heals more toxin damage (2->5 every 5 seconds).
- Increased mind control limit on the Species gland from 5 to 7 (still lasts 30 seconds).

## Why It's Good For The Game

More variety, more shenanigans. There is an annoying gland, a neutral gland and a beneficial gland, which should not skew the gland odds overall if picking randomly.
The mind control changes allow for more wacky tactics when attempting a mind control strategy: if you plant a mindshock gland and manage to crowd a lot of people into one room, you can get a huge mass of people following your orders. I'm considering adding more variants on mind control, and i have to figure out a way to convey this information to abductors; in the meantime i'll make sure to add it to the wiki once this is merged.

## Changelog
:cl: XDTM
add: Added a new abductor gland that randomizes blood type into a random reagent periodically.
add: Added a new abductor gland that links to other people, and swaps their position with the owner. Mind control on this gland will affect the linked person.
add: Added a new abductor gland that grants all access to the owner.
fix: Abductor scientists can now properly see the true name of glands when examining them.
add: Mind control on mindshock glands now "broadcasts" the mind control, affecting bystanders but not the abductee.
tweak: Species gland now has 7 mind control uses, instead of 5 (duration is still 30 seconds).
/:cl:

### CODE REVIEW

**Comment by HomocidlePancakes** [comment]
Can't wait to milk myself for initropidril huehue

**Comment by habader** [comment]
you really don't need to teach anyone anything through code, just add it and people'll learn

**Comment by CRITAWAKETS** [comment]
Does the blood reagent change react inside of your body? If it does, thats stupid and would kill you in literally every single case that you dont get blood as the reagent so i imagine it doesnt.

Also the access gland should give you an access intrinsic at random instead of straight up all access.

**Comment by XDTM** [comment]
@habader a large part of players dont keep track of all code or wiki changes. The more players know about an interesting mechanic, the more likely it will be used, which is what i want.

@CRITAWAKETS the blood change mechanic is the same that makes slime blood actually slime jelly; the reagent does not affect the gland owner, and instead injecting that reagent into them will just turn it into blood volume.
As for the access gland, i felt like all access would lead to more fun, since rotating access would, 90% of the time, leave you stuck in a department you no longer have access to until you get that access back. Plus there already is a ventcrawling gland with similar access bypassing power.

**Comment by FantasticFwoosh** [comment]
Given now you can powergame glands, why at all wouldn't abductors put all access glands / the strongest ones into themselves?

**Comment by XDTM** [comment]
never actually thought about that, maybe i should make them not work on the abductor species

**Comment by ATH1909** [comment]
> never actually thought about that, maybe i should make them not work on the abductor species

Eh, they already have the ventcrawling one and the other beneficial glands, I say leave it.

**Comment by Michael-Ashfield** [comment]
Or how about simply making it so Abductors can't put glands in themselves, after all they are meant to be experimenting on the crew, not eachother.

**Comment by coiax** [comment]
@XDTM Pretty simple way to enforce the fact the abductors can't use glands.

Glands are replacement hearts.

Abductors do not have hearts.

**Comment by coiax** [comment]
Whether a no-heart species can still be abducted and have a gland shoved in them (and whether it grants points) is up to you, but you can probably just say that the glands don't activate in a species that doesn't have use for a heart.

**Comment by XDTM** [comment]
I'm pretty sure you already cant perform the experimental surgery on heartless species anyway, i'll have to doublecheck that organ manipulation cant bypass that.

**Comment by coiax** [comment]
AND I had another thought. You should have Muscled Veins disable gland effects, given the heart becomes redundant if your veins just pump blood on their own?

**Comment by XDTM** [comment]
I'd say no for two reasons: the muscled veins surgery doesn't stop the heart, it simply gives a backup mechanism if it fails; and glands' effects are not necessarily tied to them pumping blood.

File: code/modules/antagonists/abductor/equipment/gland.dm [hunk 1]
```diff
@@ -253,6 +313,54 @@
...
+		entangled_mob = null
+
+/obj/item/organ/heart/gland/quantum/mind_control(command, mob/living/user)
+	if(..())
+		if(entangled_mob && ishuman(entangled_mob) && (entangled_mob.stat < DEAD))
+			to_chat(entangled_mob, "<span class='userdanger'>You suddenly feel an irresistible compulsion to follow an order...</span>")
+			to_chat(entangled_mob, "<span class='mind_control'>[command]</span>")
+			var/obj/screen/alert/mind_control/mind_alert = entangled_mob.throw_alert("mind_control", /obj/screen/alert/mind_control)
+			mind_alert.command = command
+			log_admin("[key_name(owner)] mirrored an abductor mind control message to [key_name(entangled_mob)]: [command]")
```
Comment (AnturK) [inline code/modules/antagonists/abductor/equipment/gland.dm]: Why log_admin this is game function ?

File: code/modules/antagonists/abductor/equipment/gland.dm [hunk 2]
```diff
@@ -170,14 +173,71 @@
...
+			continue
+		var/mob/living/carbon/human/H = M
+		if(H.stat)
+			continue
+		
+		broadcasted_mobs += H
+		to_chat(H, "<span class='userdanger'>You suddenly feel an irresistible compulsion to follow an order...</span>")
+		to_chat(H, "<span class='mind_control'>[command]</span>")
+		
+		log_admin("[key_name(user)] broadcasted an abductor mind control message from [key_name(owner)] to [key_name(H)]: [command]")
```
Comment (AnturK) [inline code/modules/antagonists/abductor/equipment/gland.dm]: same here.

File: code/modules/antagonists/abductor/equipment/gland.dm (hunk 1 above)
Comment (XDTM) [reply code/modules/antagonists/abductor/equipment/gland.dm]: must've used the wrong proc, i meant to use the one to warn admins (like with explosions)

**Pull request merged**